packages/expect/src/types.ts

```diff
@@ -70,34 +71,273 @@ export type Expect = {
 
   any(expectedObject: any): AsymmetricMatcher;
   anything(): AsymmetricMatcher;
-  arrayContaining(sample: Array<any>): AsymmetricMatcher;
-  objectContaining(sample: Record<string, any>): AsymmetricMatcher;
+  arrayContaining(sample: Array<unknown>): AsymmetricMatcher;
+  objectContaining(sample: Record<string, unknown>): AsymmetricMatcher;
   stringContaining(expected: string): AsymmetricMatcher;
   stringMatching(expected: string | RegExp): AsymmetricMatcher;
   [id: string]: AsymmetricMatcher;
   not: {[id: string]: AsymmetricMatcher};
 };
 
-type resolvesFn = {
-  [id: string]: PromiseMatcherFn;
-} & {
-  not: {[id: string]: PromiseMatcherFn};
-};
+interface Constructable {
+  new (...args: Array<unknown>): unknown;
+}
 
-type rejectsFn = {
-  [id: string]: PromiseMatcherFn;
-} & {
-  not: {[id: string]: PromiseMatcherFn};
-};
+// This is a copy from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/de6730f4463cba69904698035fafd906a72b9664/types/jest/index.d.ts#L570-L817
+export interface Matchers<R> {
+  /**
+   * Ensures the last call to a mock function was provided specific args.
+   */
+  lastCalledWith(...args: Array<unknown>): R;
+  /**
+   * Ensure that the last call to a mock function has returned a specified value.
+   */
+  lastReturnedWith(value: unknown): R;
+  /**
+   * If you know how to test something, `.not` lets you test its opposite.
+   */
+  not: Matchers<R>;
+  /**
+   * Ensure that a mock function is called with specific arguments on an Nth call.
+   */
+  nthCalledWith(nthCall: number, ...params: Array<unknown>): R;
```

Comment:
Does it help consistency to replace `...params` with `...args` here and some other occurrences later in file?